### DISCUSSION TITLE
Publishing of dependent package in workspaces project

### DISCUSSION THREAD

**Discussion by kristiannotari** [npm/feedback#751] [CLI]
**Problem Statement**

I have a npm v8 monorepo where I'm using npm workspaces. So I have 3 packages A, B, C. I want to publish package C which has packages A and B as dependencies.

From an in code import perspective, having a single node_modules let me import A and B from C code easily. But what if I want to publish C? How could the publish process work in order to "bundle" my non published package A and B respecting their respective dependencies too?

**Potential Solution**

I don't have one and I'm looking to find answers on how it is done generally as I can't find any docs on how to approach a similar problem.

**Comment by Abdillah** [comment]
@kristiannotari I'm here looking for an answer that you might have, I'm so sorry for OOT-ly replying this thread.

I was curious about you

> .. having a single node_modules let me import A and B from C code easily.

How to do that?

I was trying to install A from C but some sort of weird error happened. Do you just directly use the package? Or do you install it in some sort of way? If possible please show me the dependencies version entry after the install.

https://stackoverflow.com/questions/74412570/cross-reference-npm-install-package-inside-workspace

**Reply by kristiannotari** [reply]
Hi! From my understanding, if you have package A as a workspace inside your monorepo and you also have package B as a workspace, then you can add `"A": "<version of A in its package.json here>"` to B dependencies in `package.json` to have `npm install` actually installing the local version of A linking it from the single `node_modules` folder at the root of your monorepo.

I did not use `npm install <package A here>` via command, just wrote it in the package.json and then used `npm install`. Don't know if that makes a difference or not.

**Reply by Abdillah** [reply]
> .. actually installing the local version of A linking it from the single node_modules

Thanks!
I'm enlightened now: cross workspace package dependency is using link.

>  Don't know if that makes a difference or not.

I think that might not differ if you owned the package. In my case, I fork the package and put it in my workspace, hope that the installed package is from the workspace rather than from NPM.

Regard

**Reply by kristiannotari** [reply]
It should be (from your workspace) if you match your workspace package name and version. Otherwise it should be a bug I guess
